core-ui/src/components/Predefined/List/Services.list.js

```diff
@@ -1,29 +1,41 @@
 import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { intersperse } from 'react-shared';
 
 export const ServicesList = ({ DefaultRenderer, ...otherParams }) => {
-  const getEndpoints = service => {
-    if (service.spec.ports?.length) {
-      return service.spec.ports.map(port => {
-        const portValue = `${service.metadata.name}.${otherParams.namespace}:${port.port} ${port.protocol}`;
-        return <li key={portValue}>{portValue}</li>;
-      });
+  const { t } = useTranslation();
+
+  const getPortString = port => {
+    if (port.port === port.targetPort) {
+      return `${port.port}/${port.protocol}`;
     } else {
-      return '';
+      return `${port.port}:${port.targetPort}/${port.protocol}`;
     }
   };
 
+  const getPorts = service =>
+    intersperse(service.spec.ports?.map(getPortString) || [], ', ');
+
+  const getExternalIPs = service =>
+    intersperse(service.spec.externalIPs || [], ', ');
```

Comment:
Wouldn't using the `Array<string>::join(', ')` be just enough? I know one evaluates to an array (of strings in this case), the `join` returns a single string, but for rendering on UI the result will be the same.

Reply:
technically it came from a few ideas along the way... didn't feel like getting rid of intersperse - could be useful in the future :)